File: MMDB/Movie.m
//
//  Movie.m
//  MMDB
//
//  Created by Rob Wyland on 8/13/15.
//  Copyright (c) 2015 Org Inc. All rights reserved.
//

#import "Movie.h"

@implementation Movie

- (NSString *)overview {
    return _overview != nil && ![_overview isEqual:[NSNull null]] ? _overview : @"No Description Available";
}

- (NSString *)year {
    // Just do a quick substring since year will always be the first four chars
    return self.release_date != nil && ![self.release_date isEqual:[NSNull null]] && self.release_date.length ? [NSString stringWithFormat:@"(%@)", [self.release_date substringToIndex:4]] : nil;
}

- (NSString *)duration {
    // Do some quick calculations to determine hours and minutes
    return self.runtime != nil && ![self.runtime isEqual:[NSNull null]] ? [NSString stringWithFormat:@"%ihr %imin", (self.runtime.intValue / 60), (self.runtime.intValue % 60)] : nil;
}

- (NSString *)rating {
    // Write out the rating with a single decimal point
    return self.vote_average != nil && ![self.vote_average isEqual:[NSNull null]] ? [NSString stringWithFormat:@"%.1f", self.vote_average.floatValue] : @"";
}

@end

File: MMDB Swift/Movie.swift
//
//  Movie.swift
//  MMDB Swift
//
//  Created by Rob W on 12/11/15.
//  Copyright © 2015 Org Inc. All rights reserved.
//

import UIKit

class Movie: NSObject {
  
  dynamic var id: Int = 0
  dynamic var title: String?
  dynamic var poster_path: String?
  dynamic var release_date: String?
  dynamic var runtime: Int = 0
  dynamic var overview: String = "No Description Available"
  dynamic var vote_average: Double = 0.0
  dynamic var vote_count: Int = 0
  
  func year() -> String? {
    if let date = self.release_date where date.characters.count > 4 {
      return "(\(date.substringToIndex(date.startIndex.advancedBy(4))))"
    }
    return nil
  }
  
  func duration() -> String? {
    //if let time = self.runtime {
      return "\(self.runtime / 60)hr \(self.runtime % 60)min"
    //}
    //return nil
  }
  
  func rating() -> String? {
    //if let average = self.vote_average where self.vote_average > 0 {
      return String(format: "%.1f", self.vote_average)
    //}
    //return nil
  }
  
  override func doesNotRecognizeSelector(aSelector: Selector) {
    print("Do not know selector: \(aSelector)")
    // do nothing to avoid raising exceptions during KVO
  }
}